frontend-react/src/components/ManagePublicKey/ManagePublicKey.test.tsx

```diff
@@ -0,0 +1,126 @@
+import { screen, fireEvent } from "@testing-library/react";
+
+import { renderApp } from "../../utils/CustomRenderUtils";
+import { UseOrganizationSendersResult } from "../../hooks/UseOrganizationSenders";
+import * as useOrganizationSendersExports from "../../hooks/UseOrganizationSenders";
+import { RSSender } from "../../config/endpoints/settings";
+
+import { ManagePublicKey } from "./ManagePublicKey";
+
+const DEFAULT_SENDERS: RSSender[] = [
+    {
+        name: "default",
+        organizationName: "ignore",
+        format: "CSV",
+        customerStatus: "inactive",
+        schemaName: "primedatainput/pdi-covid-19",
+        processingType: "sync",
+        allowDuplicates: true,
+        topic: "covid-19",
+    },
+    {
+        name: "ignore-full-elr",
+        organizationName: "ignore",
+        format: "HL7",
+        customerStatus: "active",
+        schemaName: "strac/strac-covid-19",
+        processingType: "sync",
+        allowDuplicates: true,
+        topic: "full-elr",
+    },
+];
+
+describe("ManagePublicKey", () => {
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+
+    function mockUseOrganizationSenders(
+        result: Partial<UseOrganizationSendersResult> = {}
+    ) {
+        jest.spyOn(
+            useOrganizationSendersExports,
+            "useOrganizationSenders"
+        ).mockReturnValue({
+            isLoading: false,
+            senders: DEFAULT_SENDERS,
+            ...result,
+        });
+    }
+
+    describe("on load", () => {
+        describe("when more than one sender", () => {
+            beforeEach(() => {
+                mockUseOrganizationSenders({
+                    isLoading: false,
+                    senders: DEFAULT_SENDERS,
+                });
+
+                renderApp(<ManagePublicKey />);
+            });
+
+            test("renders ManagePublicKeyChooseSender", () => {
+                expect(
+                    screen.getByText(/Manage Public Key/)
+                ).toBeInTheDocument();
```

Comment:
Let's stick to `toBeVisible` over `toBeInTheDocument` since the former is stricter about actual visibility (well, as much as style-less, in-memory rendering can be).